src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1243,6 +1244,41 @@ namespace Partition {
     }
 } // namespace Partition
 
+namespace SymbolicSymbol {
+
+    static inline ASR::expr_t *eval_SymbolicSymbol(Allocator &al, const Location &loc,
+            Vec<ASR::expr_t*> &args) {
+        // TODO
+        return nullptr;
```

Comment:
My take on `eval_SymbolicSymbol` would be that we need to return a compile time value through SymEngine here . So something like this I guess 
```
    static inline ASR::expr_t *eval_SymbolicSymbol(Allocator &al, const Location &loc,
            Vec<ASR::expr_t*> &args) {
        if (args.size() == 1 && ASR::is_a<ASR::Character_t>(*args[0])) {
            ASR::Character_t* symbol_node = ASR::down_cast<ASR::Character_t>(args[0]);
            std::string symbol_name = symbol_node->m_name;
            SymEngine::Symbol symbol(symbol_name);
            // TODO: Use the symbol object for further evaluation
        }
        return nullptr;
    }
 ```
Then we might need an expr node to hold the SymEngine object !?

Reply:
Or we could do whatever operations we want here through SymEngine and finally return the value as a string !?